src/performance.js

```diff
@@ -58,8 +66,48 @@ export class Performance {
 
     /** @const @private {!Array<TickEventDef>} */
     this.events_ = [];
+
+    /** @private {boolean} */
+    this.isViewerVisible_ = false;
+
+    this.win.addEventListener('unload', this.flush.bind(this));
   }
 
+  /**
+   * Gets primary amp core services.
+   * @return {!Performance}
+   */
+  init() {
```

Comment:
@dvoytenko ah ok, i think the reason why it wasnt straight forward here was that i was using a `mock window` instead of a real `window`. ill switch to that